api/services/ocean/spark/parameters/query/deleteAction.yaml

```diff
@@ -0,0 +1,10 @@
+in: query
+name: action
+required: false
```

Comment:
I don't think there should be this parameter. 

A user wants to interrupt the application, it doesn't matter to him whether the application is currently pending, or running. This is an implementation detail for us - if the app is pending, the implementation is different than if the app is running. 

Why do we need this parameter and what does it mean that this parameter is not required? What happens if the user doesn't specify it / what happens if the user does specify it?

Reply:
This is because we are using the same http semantic e.a a `DELETE` request for both killing the application and deleting it.

Calling the delete endpoint `DELETE /ocean/spark/cluster/:clusterId/app/:appId` will default to killing the application.
Adding the action parameter will give the user or ui the ability to just go ahead and delete it without killing it.

Reply:
What do you mean by "deleting the app without killing it"?

Reply:
From my point of view, the distinction between kill and delete is that the word "kill" means that the application was alive (running), while "delete" is for an application that was about to be created (pending). So the implementation is different. 

But it's not possible to "delete the app without killing it". If you delete an app, it should go from "PENDING" to "KILLED" state. If you kill an app, it should go from "RUNNING" to "KILLED" state.

Reply:
I meant really that we just issue a `soft-delete` in the db and marking it as deleted.

I guess
>  "deleting the app without killing it"
Is really an implementation detail. This path has not been implemented yet and I was actually just making an assumption as to how it should work, it's still very much up for disussion.

Reply:
Ok so if the distinction is  just dependent on the state of the app then the query parameter is completely unneeded and we should remove it.

Reply:
Discussed offline (on Slack huddle). 

Let's not document this flag, for now the only user-facing behavior is "Killing". 

In the future, maybe we could implement a delete action whick kills the application and also marks the application as deleted (soft-delete) so it's not visible anymore in the dashboard. But we can document this when get there!